api4/role.go

```diff
@@ -162,6 +162,11 @@ func patchRole(c *Context, w http.ResponseWriter, r *http.Request) {
 				return
 			}
 		}
+
+		ancillaryPermissions := model.AddAncillaryPermissions(*patch.Permissions)
+		*patch.Permissions = append(*patch.Permissions, ancillaryPermissions...)
```

Comment:
@hahmadia This change makes prevents one from removing ancillary permissions via MMCtl.

Reply:
@mkraft 
This was so we move the Ancillary permission logic from front end to the backend so that we don't have to keep a copy of them on frontend and have to update them when we update in the backend.

Recall:
https://github.com/mattermost/mattermost-server/pull/17061
https://github.com/mattermost/mattermost-webapp/pull/7628

Reply:
We were determining what ancillary permissions are needed on the front-end using the list of ancillary permission mappings in redux here
https://github.com/mattermost/mattermost-redux/pull/1392

The point was not to have to keep that list because every time we added or removed ancillary in the backend, we needed to update it in the frontend too or else we stopped getting the ancillary permissions we required.

Reply:
I totally agree that it's better for that to live on the backend. My main concern is that when you attempt to remove a permission it not fail silently (and you only know about it by checking the database). I think it should probably remove the permission. 

That said I totally understand wanting to automatically add ancillary permissions when you add a sysconsole permission.

Also, you may want to consider removing ancillary permissions when a sysconsole permission is removed. That's what MMCtl does for each ancillary permission that the sysconsole permission used, if they're not still require by other sysconsole permissions.